brabbuss/refactor-tractor
# Feature asynch pageload

### What does this PR do?
(*please link commit if helpful by copy/paste commit id*)
1. Refactors code in scripts.js.  Takes global, data-dependent code that was invoking on page load and wraps it in a helper function.  The helper function is located inside a .then(), which is attached to Promise.all.  The result is that none of this code is invoked until after the data promises are resolved.

### What are the relevant tickets (please link)?
1.  Issue #4 

### Where should the reviewer start?
1. Line 92-224 (scripts.js)

### How should this be tested?
1. You can see that the data is populating on the page in real time.  You can also console.log variables to see that they are there.

### Any background context you want to provide?
1.  What is in this PR will still require additional refactoring.  Functionality is broke down into five sub-topics, but each is still very long.

### Screenshots (if applicable)
1.

### New Questions/Thoughts?:
1.  Now that we have this working, we can start refactoring the sub-functionality for greater SRP.
2. We need to discuss the variable **todayDate**.  It is set to a date much older than the data, but when I tried updating it the code was broken.
